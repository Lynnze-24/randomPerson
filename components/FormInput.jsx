import React,{useState} from 'react'
import {  Text,StyleSheet } from 'react-native'
import { TextInput } from 'react-native-paper'
import cusTheme from '../theme/Theme'

const FormInput = ({value,changeValue,testID,label,hasError,errorMessage,onKeyUp,isPassword}) => {
    const [isPwVisible, setIsPwVisible] = useState(true);
    const togglePasswordInput = ()=> isPwVisible? setIsPwVisible(false): setIsPwVisible(true)
    
    return (<>
        <TextInput 
            
             label={label}
             secureTextEntry={isPassword && isPwVisible?true:false}
            dense={true}
            mode='outlined'
            onKeyUp={onKeyUp}
            style={styles.input}
            value={value}
            outlineColor={hasError?cusTheme.danger:cusTheme.primary} 
            onChangeText={changeValue}
            activeOutlineColor={hasError?cusTheme.danger:cusTheme.primary} 
            right={isPassword && (<TextInput.Icon testID ="pwToggle" name="eye" onPress={togglePasswordInput} />)}
            testID={testID} 
            />
           { hasError && (<Text style={styles.errorText}>{errorMessage}</Text>)}
            </>
    )
}


const styles = StyleSheet.create({
    input:{
        width:350,
        maxWidth:'100%',
        marginTop:10,
        fontSize:18
    },
    errorText:{
        textAlign:'left',
        color:cusTheme.danger
    }
});



export default FormInput
