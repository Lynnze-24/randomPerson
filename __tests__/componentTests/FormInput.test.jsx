import {render, fireEvent } from "@testing-library/react-native";
import FormInput from "../../components/FormInput";
import renderer from 'react-test-renderer';



it('matches snapShot', () => {
    const comp = renderer.create(<FormInput testID="test"/>).toJSON();
    expect(comp).toMatchSnapshot();
});

it('should render component correctly',()=>{ 
    const {getByTestId}=render(<FormInput testID="test"/>);
    getByTestId("test")
})

it('should show error message if hasError is true',()=>{ 
    const {getByText}=render(<FormInput errorMessage = "Invalid Username" hasError={true} testID="test"/>);
    getByText( "Invalid Username")
})

it('should not show error message if hasError is false',()=>{ 
    const {queryByText}=render(<FormInput errorMessage = "Invalid Username" hasError={false} testID="test"/>);
    expect(queryByText("Invalid Username")).toBe(null)
})

it('should hide text initially if isPassword is true',()=>{ 
    const {getByTestId,queryByText,debug}=render(<FormInput isPassword={true} testID="test"/>);
   
    const textInput = getByTestId("test");
    expect(textInput).toHaveProp("secureTextEntry",true)
})

it('should switch password visibility when toggle eye button',()=>{ 
    const {getByTestId,queryByText,debug}=render(<FormInput isPassword={true} testID="test"/>);
    const textInput = getByTestId("test");
    expect(textInput).toHaveProp("secureTextEntry",true)
     fireEvent.press(getByTestId("pwToggle")) 
     expect(textInput).toHaveProp("secureTextEntry",false)
})