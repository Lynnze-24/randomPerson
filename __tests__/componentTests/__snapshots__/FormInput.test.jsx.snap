// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<View
  style={
    Object {
      "marginTop": 10,
      "maxWidth": "100%",
      "width": 350,
    }
  }
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 6,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "borderColor": "#120311",
          "borderRadius": 4,
          "borderWidth": 1,
        },
      ]
    }
    testID="text-input-outline"
  />
  <View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
          },
          Object {
            "minHeight": 48,
            "paddingTop": 8,
          },
        ]
      }
    >
      <TextInput
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        secureTextEntry={false}
        selectionColor="#120311"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "paddingHorizontal": 14,
              "zIndex": 1,
            },
            Object {
              "height": 40,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 18,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        testID="test"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
